crates/but-github/src/lib.rs

```diff
@@ -242,6 +251,25 @@ pub async fn get_gh_user(
     }
 }
 
+/// Check if an error is a network connectivity error.
+///
+/// This includes DNS resolution failures, connection timeouts, connection refused, etc.
+fn is_network_error(err: &anyhow::Error) -> bool {
+    // Check if it's a ReqwestError wrapped in ClientError
+    if let Some(octorust::ClientError::ReqwestError(reqwest_err)) =
+        err.downcast_ref::<octorust::ClientError>()
+    {
+        return reqwest_err.is_timeout() || reqwest_err.is_connect() || reqwest_err.is_request();
+    }
+
+    // Fallback: check for reqwest::Error directly in the error chain
+    err.chain().any(|cause| {
+        cause
+            .downcast_ref::<reqwest::Error>()
+            .is_some_and(|e| e.is_timeout() || e.is_connect() || e.is_request())
+    })
```

Comment:
Downcast actually takes care of walking the error chain, so this can be removed entirely.